### PR TITLE
FORMS-13810: Add the field to allow custom message for format in date…

### PR DESCRIPTION
**Pull request by s1mahanty**
… picker

## Related Issue
[FORMS-13810](https://jira.corp.adobe.com/browse/FORMS-13810)

## Screenshots (if appropriate):
When no custom message provided
Screenshot 2024-04-03 at 12 36 13 PM

Added a field to add custom message for format

Screenshot 2024-04-03 at 1 10 42 PM

After custom message is added
Screenshot 2024-04-03 at 1 10 58 PM

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

**Review by rismehta — CHANGES_REQUESTED** [review]
Please write a runtime test case for this as part of datapicker spec so that the use-case works end to end

File: ui.tests/test-module/specs/datepicker/datepicker.runtime.spec.js [hunk 1]
```diff
@@ -256,4 +256,15 @@ describe("Form Runtime with Date Picker", () => {
...
     });
 
+    it(" Test custom error message when incorrect date format is entered", () => {
+        const [datePicker5, datePicker5FieldView] = Object.entries(formContainer._fields)[6];
+        const incorrectInput = "01-01-2023";
+
+        cy.get(`#${datePicker5}`).find("input").should('have.attr',"type", "text");
+        cy.get(`#${datePicker5}`).find("input").clear().type(incorrectInput).blur().then(x => {
+            cy.get(`#${datePicker5}`).find(".cmp-adaptiveform-datepicker__errormessage").should('have.text',"Date format expected is d/M/y")
+        })
```
Comment (vdua) [inline ui.tests/test-module/specs/datepicker/datepicker.runtime.spec.js:266]: can you please add the test case where the date itself is invalid (the following values)
```
const incorrectInput = 'adfasdfa'
const incorrectInput = '29/2/2023'
const incorrectInput = '1/13/2023'
const incorrectInput = '32/1/2023'
```

Will it show the error message

File: it/content/src/main/content/jcr_root/content/forms/af/core-components-it/samples/datepicker/basic/.content.xml [hunk 2]
```diff
@@ -114,10 +115,11 @@
                     editPatternType="d/M/y"
                     enabled="{Boolean}true"
                     fieldType="date-input"
+                    formatMessage="Date format expected is d/M/y"
                     hideTitle="false"
                     name="datepicker7"
                     readOnly="{Boolean}false"
-                    textIsRich="[true,true]"
+                    textIsRich="[true,true,true]"
```
Comment (sakshi-arora1) [inline it/content/src/main/content/jcr_root/content/forms/af/core-components-it/samples/datepicker/basic/.content.xml:122]: this property shouldn't be there

**Review by sakshi-arora1 — APPROVED** [review]
Just a minor comment on collateral

**Pull request merged**